plugins/volume/org.xen.xapi.storage.zfs-ng/sr.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+
+import os
+import os.path
+import sys
+import urlparse
+
+from xapi.storage import log
+from xapi.storage.libs import util
+from xapi.storage.libs.libcow.zfsutil import ZFSUtil
+from xapi.storage.common import call
+from xapi.storage.libs.libcow.imageformat import ImageFormat
+from xapi.storage.libs.libcow.volume import COWVolume
+from xapi.storage.libs.libcow.callbacks import VolumeContext
+import xapi.storage.api.v5.volume
+
+import importlib
+
+ZPOOL_BIN = 'zpool'
+ZFS_BIN = 'zfs'
+
+@util.decorate_all_routines(util.log_exceptions_in_function)
+class Implementation(xapi.storage.api.v5.volume.SR_skeleton):
+    def probe(self, dbg, configuration):
+        return {
+            'srs': [],
+            'uris': []
+        }
+
+    def attach(self, dbg, configuration):
+        uri = configuration['mountpoint']
+        log.debug('{}: SR.attach: config={}, uri={}'.format(
+            dbg, configuration, uri))
+
+        sr = urlparse.urlparse(uri).path
+        return sr
+
+    def create(self, dbg, sr_uuid, configuration, name, description):
+        log.debug('{}: SR.create: config={}, sr_uuid={}'.format(
+            dbg, configuration, sr_uuid))
+
+        if 'devices' not in configuration:
+            log.error('devices parameter is missed')
+            raise
+        devs = configuration['devices'].split(',')
+
+        mountpoint = '/' + name
+        if 'mountpoint' in configuration:
+            mountpoint = configuration['mountpoint']
```

Comment:
Or:

```python
configuration.get('mountpoint', '/' + name)
```

Reply:
If line 48 is True, the mountpoint is got from configuration otherwise we use the value sets at line 47. Is it OK like this or an else would it make it clear?